cardano-api/test/Test/Cardano/Api/MetaData.hs

```diff
@@ -1,39 +1,54 @@
 {-# LANGUAGE TemplateHaskell #-}
 {-# LANGUAGE TupleSections #-}
+{-# LANGUAGE OverloadedStrings #-}
 module Test.Cardano.Api.MetaData
   ( tests
   ) where
 
 import           Cardano.Prelude hiding (MetaData)
 
-import           Cardano.Api.Typed
 import           Cardano.Api.MetaData
+import           Cardano.Api.Typed
+import qualified Shelley.Spec.Ledger.MetaData as MD
 
+import qualified Data.ByteString.Base16 as Base16
 import qualified Data.ByteString.Char8 as BS
-import qualified Data.Map.Strict as Map
+import           Data.Either (isLeft, fromRight)
 import qualified Data.List as List
+import qualified Data.Map.Strict as Map
 import qualified Data.Text as Text
+import qualified Data.Text.Encoding as Text
 
 import           Hedgehog (Gen, Property, discover)
 import qualified Hedgehog
 import qualified Hedgehog.Gen as Gen
-import           Hedgehog.Range (Range)
+import qualified Hedgehog.Internal.Gen as Gen
 import qualified Hedgehog.Range as Range
 
 -- This test is fragile to changes in the way Metadata is converted to/from JSON.
 prop_round_trip_json_metadatum :: Property
 prop_round_trip_json_metadatum = do
   Hedgehog.withTests 1000 . Hedgehog.property $ do
     md <- Hedgehog.forAll genMetaData
-    Hedgehog.tripping md jsonFromMetadata jsonToMetadata
-
--- -----------------------------------------------------------------------------
+    -- pre-wash metadata to remove ambiguous metadatums before
+    -- testing the round-trip property.
+    let md' = fromRight md $ jsonToMetadata $ jsonFromMetadata md
```

Comment:
On second thoughts, this is going to make the round-trip test much less effective...

Reply:
Ideally, I'd rather generate arbitrary JSON values and see that they roundtrip fine using the metadata conversion function, it'll better capture what is the actual property we want. By re-using jsonToMetadata and jsonFromMetadata to construct the input I am afraid we are indeed creating a bias in the generator which will fail to capture some edge cases.